### DISCUSSION TITLE
About some UI config elements

### DISCUSSION THREAD

**Discussion by NullPlane** [SpartanJ/ecode#285] [Ideas]
Randomly stumbled upon this:
https://github.com/SpartanJ/ecode/issues/261
and I had an idea. Why not tie all
configs to a single interface.

As there was a multitude of ideas, e.g. [translation helper](https://github.com/SpartanJ/ecode/issues/122#issuecomment-1984962849 "#122 Multi Human Language Support")
I wondered, whether it might be beneficial to have
an additional config manager as a custom element
like the CSS manager or as a plugin. It might be 
fitted with a layout that may show keys or switches
or objects as some sort of table or dropdown list
and if applicable may also be fitted with its own 
translations. After that it's just to be filled it with parsed 
elements from some configs like the i18n-XML or the 
config files for other plugins. So there'd be one Element 
that is versatile and can be fed to manipulate quite a few 
things to choose from, the work for individual problems
is done in the background individually depending on 
what's chosen. 

In case more third party plugins would be planned (idk) this 
might also provide them with a kind of surface to interface
with their custom problems as they may tie in their plugins
to that manager. Maybe they could include documentation
there as a kind of listed browsing structure.

Just an idea but to me it is questionable if it'd be worth a 
tussle. I may guess tho if more people get to know the editor
it'd be of use for some.

**Comment by SpartanJ** [comment]
Hi!
I'm having a hard time following what you're trying to describe, maybe could you give me a concrete example?

> Why not tie all configs to a single interface.

You mean user interface or single settings file?

Sorry, I'm lost. If you refer to the UI [this](https://github.com/SpartanJ/ecode/issues/279#issuecomment-2112621974) might give you some hint on what I'm planning. I would like to add an easy descriptive format to generate the settings panel but that might take some time to develop.

**Comment by NullPlane** [comment]
Okay, yeah. Essentially something like in this example for Obsidian was what I thought. Dynamically generated sections. Each plugin has their buttons.

![temp](https://github.com/SpartanJ/ecode/assets/111517889/1aa91ab3-7d9c-45b7-816c-efe4977ffae4)

The only difference between that was to allow for parsing the default used formats in the editor (XML [i18n], JSON, CFG [Editor, Colorschemes]) but if you have a lib to convert that from an unifying format that is fine.

Essentially in there I thought - especially for plugins etc. - that besides a classical toggle-interface like in any settings menu there'd be a possibility to have some kind of Table-View for the configs requiring default-files with all possible keys and documentative fields. Maybe something toggleable to switch to in the UI. The cells may contain keys or formatted and recognized comments, which provide documentative information to later be shown. The whole view gives ability for manipulation and an overview.

So I thought the translation manager and checker may be integrated there somewhere.

---

This is very very roughly what I thought of but it'd have an information column and would be a little more usable than this concept application. This could - for the UI part with the switches - be fed into hovering hints or little `ℹ️`-infosymbols or whatever. The example is JSON but I thought for something overarching for the configs.

![Screenshot 2024-06-27 010610](https://github.com/SpartanJ/ecode/assets/111517889/db7b6f85-a4fa-44d8-8322-1a2fd755a814)

For simplicity I did just add two example hints and it won't work out as I'd want to show it but in the end the Creator of each plugin or whatever may include more hints to all their sub-entries.
